clippy_lints/src/import_rename.rs

```diff
@@ -0,0 +1,99 @@
+use clippy_utils::{diagnostics::span_lint_and_sugg, source::snippet_opt};
+
+use rustc_data_structures::fx::FxHashSet;
+use rustc_errors::Applicability;
+use rustc_hir::{def::Res, def_id::DefId, Crate, Item, ItemKind, UseKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_tool_lint, impl_lint_pass};
+use rustc_span::Symbol;
+
+declare_clippy_lint! {
+    /// **What it does:** Checks that any import matching `import_renames` is renamed.
+    ///
+    /// **Why is this bad?** This is bad for some items because their defined names are
+    /// too vauge outside of their defining scope.
+    ///
+    /// **Known problems:** Any import must use the original fully qualified path.
+    ///
+    /// For example, if you want to rename `serde_json::Value`, your clippy.toml
+    /// configuration would look like
+    /// `import-renames = [ { path: "serde_json::value::Value", rename = "JsonValue" }]` and not
+    /// `import-renames = [ { path: "serde_json::Value", rename = "JsonValue" }]` as you might expect.
+    ///
+    ///
+    /// **Example:**
+    ///
+    /// ```rust,ignore
+    /// use serde_json::Value;
+    /// ```
+    /// Use instead:
+    /// ```rust,ignore
+    /// use serde_json::Value as JsonValue;
+    /// ```
+    pub IMPORT_RENAME,
+    restriction,
+    "enforce import renames"
+}
+
+pub struct ImportRename {
+    imports: FxHashSet<(Vec<Symbol>, Symbol)>,
+    def_ids: FxHashSet<(DefId, Symbol)>,
```

Comment:
```suggestion
    conf_renames: Vec<Rename>,
    renames: FxHashMap<DefId, Symbol>,
```
The first HashSet is not necessary, may as well keep the Vec. The second one would be better as a HashMap.